connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/ArtifactManager.scala

```diff
@@ -108,6 +104,56 @@ class ArtifactManager(
    */
   def addArtifact(uri: URI): Unit = addArtifacts(parseArtifacts(uri))
 
+  /**
+   * Add a single in-memory artifact to the session while preserving the directory structure
+   * specified by `target` under the session's working directory of that particular file
+   * extension.
+   *
+   * Supported target file extensions are .jar and .class.
+   *
+   * ==Example==
+   * {{{
+   *  addArtifact(bytesBar, "foo/bar.class")
+   *  addArtifact(bytesFlat, "flat.class")
+   *  // Directory structure of the session's working directory for class files would look like:
+   *  // ${WORKING_DIR_FOR_CLASS_FILES}/flat.class
+   *  // ${WORKING_DIR_FOR_CLASS_FILES}/foo/bar.class
+   * }}}
+   */
+  def addArtifact(bytes: Array[Byte], target: String): Unit = {
```

Comment:
Where is this used? I am not against it, just trying to understand?

Reply:
It's not used at the moment.  We do, however, use a similar variation of this (without a custom `target`)  in the `AmmoniteClassFinder`.  This API would let user's do the same if they wish to.